jetty-client/src/main/java/org/eclipse/jetty/client/GZIPContentDecoder.java

```diff
@@ -39,13 +47,47 @@ public GZIPContentDecoder(ByteBufferPool byteBufferPool, int bufferSize)
         super(byteBufferPool, bufferSize);
     }
 
+    @Override
+    public void beforeDecoding(HttpExchange exchange)
+    {
+        HttpFields.Mutable headers = exchange.getResponse().headers();
+        // Content-Length is not valid anymore while we are decoding.
+        headers.remove(HttpHeader.CONTENT_LENGTH);
```

Comment:
Don't do a remove here as that is an iteration and you are already iterating below with a listIterator that can remove any content-length